src/app/data/oasis-account-names.ts

```diff
@@ -40,25 +40,39 @@ const dataSources: Record<Network, Partial<Record<Layer, string>>> = {
 const getOasisAccountsMetadata = async (network: Network, layer: Layer): Promise<AccountData> => {
   const url = dataSources[network][layer]
   if (!url) throw new Error('No data available for this layer')
-  const response = await axios.get(url)
-  if (response.status !== 200) throw new Error("Couldn't load names")
-  if (!response.data) throw new Error("Couldn't load names")
-  // console.log('Response data is', response.data)
-  const map: AccountMap = new Map()
-  const list: AccountMetadata[] = []
-  Array.from(response.data).forEach((entry: any) => {
-    const metadata: AccountMetadata = {
-      address: entry.Address,
-      name: entry.Name,
-      description: entry.Description,
+
+  try {
+    const response = await axios.get(url)
+    if (response.status !== 200) throw new Error("Couldn't load names")
+    if (!response.data) throw new Error("Couldn't load names")
+    const map: AccountMap = new Map()
+    const list: AccountMetadata[] = []
+    Array.from(response.data).forEach((entry: any) => {
+      const metadata: AccountMetadata = {
+        address: entry.Address,
+        name: entry.Name,
+        description: entry.Description,
+      }
+      // Register the metadata in its native form
+      list.push(metadata)
+      map.set(metadata.address, metadata)
+    })
+    return {
+      map,
+      list,
+    }
+  } catch (error) {
+    if (axios.isAxiosError(error)) {
```

Comment:
I think ignoring errors should be handled per use - and I thought it already was:
- https://github.com/oasisprotocol/explorer/blob/12468641e89f5f06b65b6098fde88f859fed9368/src/app/data/oasis-account-names.ts#L106
- https://github.com/oasisprotocol/explorer/blob/12468641e89f5f06b65b6098fde88f859fed9368/src/app/components/Account/AccountLink.tsx#L93

Reply:
this is weird, but Ok. our pattern is "disable suspense's default behavior of throwing errors to the error boundary" and we do it per use like here https://github.com/oasisprotocol/explorer/blob/master/src/app/data/oasis-account-names.ts#L104.

Reply:
even worse:
- react-query's default behavior is: don't throw error (v5 useErrorBoundary -> throwOnError)
- we change defaults to throw on 5xx, and on disconnect (because no status code, because no CORS, because cloudflare) https://github.com/oasisprotocol/explorer/blob/2955d9df60a282414575c214412a4e906891a0b7/src/index.tsx#L17-L27
- and override the default when result isn't important (e.g. coingecko price)

I'd prefer if we ignored errors inside the displaying component. E.g. transfer value UI component knows that it isn't criticial if coingecko fails. But I took a shortcut in https://github.com/oasisprotocol/explorer/commit/bc66fc4d7230b9416527d40da54f8c5d66d94c28 and ignored errors inside the hooks, because it would be too much work to refactor.

Current
```
<SearchResultsPage
  const tokenPrices =
    useAllTokenPrices() ->
      useGetTokenPricesFromGecko() ->
        // Ignore errors. hopefully every use in every parent hook and all its sub-component isn't essential
        useQuery(useErrorBoundary: false)
  <SearchResultsView tokenPrices={tokenPrices}
    <GlobalSearchResultsView tokenPrices={tokenPrices}
      <SearchResultsList tokenPrices={tokenPrices}
        <RuntimeTransactionDetailView tokenPrices={tokenPrices}
          <dd><CurrentFiatValue amount={transaction.amount} ...tokenPrices /></dd>
```

Prefer
```
<SearchResultsPage
  <SearchResultsView
    <GlobalSearchResultsView
      <SearchResultsList
        <RuntimeTransactionDetailView
          const tokenPrices = 
            // converted tx value display isn't essential, ignore errors
            useAllTokenPrices(useErrorBoundary: false) ->
              useGetTokenPricesFromGecko(useErrorBoundary) ->
                useQuery(useErrorBoundary)
          <dd><CurrentFiatValue amount={transaction.amount} ...tokenPrices /></dd>
```

But this conflicts with my other preference: most UI components should be pure, so lets move fetching towards Page components.

:shrug: